### PR TITLE
[libc] document supported os ranges

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
LLVM-libc does not guarantee backward compatibility with operating systems that have reached their EOL.
Compatibility patches for obsolete operating systems will not be accepted.

For Linux, we support kernel versions as listed on [kernel.org](https://kernel.org/), including
``longterm`` (not passed EOL date), ``stable``, and ``mainline`` versions. We actively adapt new features
from ``linux-next``.

For Windows, we plan to support products within their lifecycle. Please refer to 
[Search Product and Services Lifecycle Information](https://learn.microsoft.com/en-us/lifecycle/products/?products=windows) for more information.

### CODE REVIEW

File: libc/docs/index.rst [hunk 1]
```diff
@@ -40,6 +40,16 @@ and MacOS have limited support (may be broken).  The Fuchsia platform is
 slowly replacing functions from its bundled libc with functions from this
 project.
 
+LLVM-libc does not guarantee backward compatibility with operating systems that reaches their EOL.
+Compatibility patch for obsolete operating systems will not be accepted.
+
+For Linux, we support kernel versions as listed on `kernel.org <https://kernel.org/>`_, including
+``longterm``, ``stable``, and ``mainline`` versions. We actively adapt new features
```
Comment (nickdesaulniers) [inline libc/docs/index.rst:47]:
```suggestion
``longterm`` (not past EOL date), ``stable``, and ``mainline`` versions. We actively adopt new features
```

File: libc/docs/index.rst [hunk 2]
```diff
@@ -40,6 +40,16 @@ and MacOS have limited support (may be broken).  The Fuchsia platform is
 slowly replacing functions from its bundled libc with functions from this
 project.
 
+LLVM-libc does not guarantee backward compatibility with operating systems that reaches their EOL.
```
Comment (nickdesaulniers) [inline libc/docs/index.rst:43]:
```suggestion
LLVM-libc does not guarantee backward compatibility with operating systems that have reached their EOL.
```

File: libc/docs/index.rst [hunk 3]
```diff
@@ -40,6 +40,16 @@ and MacOS have limited support (may be broken).  The Fuchsia platform is
...
+Compatibility patch for obsolete operating systems will not be accepted.
```
Comment (nickdesaulniers) [inline libc/docs/index.rst:44]:
```suggestion
Compatibility patches for obsolete operating systems will not be accepted.
```

**Comment by nickdesaulniers** [comment]
For implications to the codebase, this means that for our syscall wrappers, we have checks that `SYS_<foo>` is defined.  If foo has been supported by all supported architectures for all versions of LTS linux, then we should not clutter the code with prepropecessor guards.

https://www.chromium.org/chromium-os/developer-library/reference/linux-constants/syscalls/ talks a little about this, but doesn't have the kernel version info. I swear on hacker news recently someone linked to a library from chromeos that was trying to do syscall wrappers. It looked interesting, but I cannot find it.

Also, we should stop looking at the host's kernel headers when building llvm-libc for linux, and always require they be passed in. Otherwise, we'll have the wrong syscall numbers when cross compiling.

**Review by nickdesaulniers — APPROVED** [review]

File: libc/docs/index.rst [hunk 4]
```diff
@@ -40,6 +40,16 @@ and MacOS have limited support (may be broken).  The Fuchsia platform is
...
 
+LLVM-libc does not guarantee backward compatibility with operating systems that have reached their EOL.
+Compatibility patches for obsolete operating systems will not be accepted.
+
+For Linux, we support kernel versions as listed on `kernel.org <https://kernel.org/>`_, including
+``longterm`` (not past EOL date), ``stable``, and ``mainline`` versions. We actively adopt new features
+from ``linux-next``.
+
+For Windows, we support products within their lifecycle. Please refer to 
+`Search Product and Services Lifecycle Information <https://learn.microsoft.com/en-us/lifecycle/products/?products=windows>`_ for more information.
```
[the remark this answers is not in the document]
Comment (SchrodingerZhu) [reply libc/docs/index.rst]: applied!

**Review by michaelrj-google — APPROVED** [review]

**Comment by petrhosek** [comment]
> https://www.chromium.org/chromium-os/developer-library/reference/linux-constants/syscalls/ talks a little about this, but doesn't have the kernel version info. I swear on hacker news recently someone linked to a library from chromeos that was trying to do syscall wrappers. It looked interesting, but I cannot find it.

https://chromium.googlesource.com/linux-syscall-support/

**Pull request merged**